allauth/socialaccount/providers/twitch/provider.py

```diff
@@ -28,7 +28,6 @@ def extract_uid(self, data):
     def extract_common_fields(self, data):
         return {
             "username": data.get("login"),
-            "name": data.get("display_name"),
```

Comment:
We cannot assume that display_name == some_sort_of_given_legal_name. 
Reference: 
https://dev.twitch.tv/docs/api/reference/#get-users

The ticket also has a screenshot of the types of errors this causes on our Platform. 

I will probably submit an upstream ticket to djanog-allauth but this internal way seemed faster for our customer's upcoming event.